crates/adapters/src/server/mod.rs

```diff
@@ -330,7 +329,12 @@ async fn shutdown(state: WebData<ServerState>) -> impl Responder {
     let controller = state.controller.lock().unwrap().take();
     if let Some(controller) = controller {
         match controller.stop() {
```

Comment:
Both the `stop` call and the actix termination command below can be slow or even get stuck in some cases. We will need a way to enforce eventual termination from the pipeline manager by killing the process/pod after a timeout.  This is outside the scope of this PR.  I will create an issue.